silkworm/core/execution/evm.hpp

```diff
@@ -72,6 +73,19 @@ using EvmTracers = std::vector<std::reference_wrapper<EvmTracer>>;
 
 using AnalysisCache = LruCache<evmc::bytes32, std::shared_ptr<evmone::baseline::CodeAnalysis>>;
 
+using TransferFunc = void(IntraBlockState& state, const evmc::address& sender, const evmc::address& recipient,
+                          const intx::uint256& amount, bool bailout);
+
+// See consensus.Transfer in Erigon
+inline void standard_transfer(IntraBlockState& state, const evmc::address& sender, const evmc::address& recipient,
+                              const intx::uint256& amount, bool bailout) {
+    // TODO(yperbasis) why is the bailout condition different from Erigon?
```

Comment:
Probably it appears to be different because in Erigon the bailout mechanism is handled all in [one module](https://github.com/erigontech/erigon/blob/main/core/state_transition.go), whilst in Silkworm it's unfortunately done in two different modules:
- in `rpc::EVMExecutor` for the [gas](https://github.com/erigontech/silkworm/blob/master/silkworm/rpc/core/evm_executor.cpp#L300)
- here in `EMV` for the value transferred

It is worth it to amend this, so I'm going open an issue on this topic.

Moreover, there's another very related topic that's worth improving as well: `rpc::EVMExecutor` captures a (simulated) EVM execution not so differently wrt `ExecutionProcessor` and exists for historical reasons (i.e. it comes from standalone Silkrpc project), but I've been having this idea for a while of getting rid of it by moving its additional functionality into `ExecutionProcessor`. I will open an issue also for this.